app/src/main/java/com/example/myapplication/Expressions/Num.java

```diff
@@ -1,12 +1,19 @@
 package com.example.myapplication.Expressions;
 
-public class Num extends Expr {
-    public float number;
-    public Num(float num) {
-        this.number = num;
+public final class Num extends UnaryOperation {
```

Comment:
Можно назвать `Number`, раз уж мы все решили полностью именовать без сокращений.

Если вы хотите записывать выражения в коде как-то более элегантно, то можете создать утилитарный класс `Functions` и сделать там фактори методы:
```
class Functions {
  public static final Argument x = new X();

  public static Number n(double value) {
    return new Number(value);
  }

  public static final mul(Function left, Function right) {
    return new Multiplication(left, right);
  }

  // And so on...
}

// So in code if you import all static members from Functions 
// it will look like this:
System.out.println(mul((n(2) + x), 2).asString()); // ((2 + x) * 2)
```

И опять же! Это не унарная операция! Литерал еще менее унарен, чем `x`!

Reply:
Неа, нельзя. Имя Number запривачено

Reply:
Ааа, точно, такой класс есть в стандартной библиотеке. Ну, можно оставить так. Еще можно назвать `Literal`, как в языках программирования называют такие типа захардкоженные значения.